addons/website/models/ir_ui_view.py

```diff
@@ -93,6 +93,54 @@ def write(self, vals):
 
         return True
 
+    @api.multi
+    def _get_specific_views(self):
+        """ Given a generic view (no website_id), return a record set containing
+            all the views that have been COW'd from it.
+        """
+        self.ensure_one()
+        domain = [('key', '=', self.key), ('website_id', '!=', False)]
+        return self.with_context(active_test=False).search(domain)
+
+    def _load_records_write(self, values):
+        """ During module update, when updating a generic view, we should also
+            update its specific views (COW'd).
+            Note that we will only update unmodified fields. That will mimic the
+            noupdate behavior on views having an ir.model.data.
+        """
+        if self.type == 'qweb' and not self.website_id:
+            # Update also specific views
+            for cow_view in self._get_specific_views():
+                authorized_vals = {}
+                for key in values:
+                    if cow_view[key] == self[key]:
+                        authorized_vals[key] = values[key]
+                cow_view.write(authorized_vals)
+            super(View, self)._load_records_write(values)
+
+    def _load_records_create(self, values):
+        """ During module install, when creating a generic child view, we should
+            also create that view under specific view trees (COW'd).
+            Top level view (no inherit_id) do not need that behavior as they
```

Comment:
Just a question here: now that we have those methods, wouldn't it be ok/better to directly create all the specific views always, and never work with the generic one, so we can get rid of COW?

Reply:
Not sure to fully understand what you mean.
This is only related to module update and install about XML files views.
Multi website and COW is much bigger that just that, mostly pages. Imagine you create a generic page, we still need the COW when the page view will be edited on a website.
You might also create a complete generic view tree yourself that has no XML id since you created it in the backend, it will need the COW behavior.
Note that copying all views for every websites would be overkill as 99% of views will never be COW'd and just a generic is enough (like grenache that has more than 10 websites).

Reply:
Thank you for the details, this answers my question. 👍